File: espritx-Symfony V4/espritx-main/src/Entity/GroupPost.php
<?php

namespace App\Entity;

use App\Repository\GroupPostRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Gedmo\Mapping\Annotation as Gedmo;
use Symfony\Component\Validator\Constraints as Assert;


/**
 * @ORM\Entity(repositoryClass=GroupPostRepository::class)
 */
class GroupPost
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @Assert\NotBlank
     * @Assert\Length(
     *      min = 3,
     *      max = 15,
     *      minMessage = "le nom de votre groupe doit avoir au moins {{ limit }} caracteres ",
     *      maxMessage = "le nom de votre groupe ne doit pas depasser  {{ limit }} caracteres"
     * )
     * @ORM\Column(type="string", length=255)
     */
    private $nomGroupe;

    /**
     * @Gedmo\Slug(fields={"nomGroupe"})
     * @ORM\Column(type="string", length=255)
     */
    private $slug;

    /**
     * @ORM\OneToMany(targetEntity=Post::class, mappedBy="groupPost",cascade={"remove"})
     */
    private $posts;

    /**
     * @Assert\NotBlank
     *
     *
     * @ORM\Column(type="string", length=255)
     */
    private $image;

    /**
     * @ORM\ManyToOne(targetEntity=User::class, inversedBy="groupPosts")
     * @ORM\JoinColumn(nullable=false)
     */
    private $user;

    /**
     * @ORM\Column(type="datetime_immutable")
     */
    private $createdAt;

    /**
     * @ORM\Column(type="boolean")
     */
    private $isValid;

    /**
     * @ORM\Column(type="boolean")
     */
    private $isDeleted;

    /**
     *@Assert\Length(
     *      min = 5,
     *      max = 2000,
     *      minMessage = "le but de votre groupe doit avoir au moins {{ limit }} caracteres ",
     *      maxMessage = "le but de votre groupe ne doit pas depasser  {{ limit }} caracteres"
     * )
     * @ORM\Column(type="text")
     */
    private $but;

    /**
     * @ORM\ManyToMany(targetEntity=User::class, inversedBy="groupes")
     */
    private $membre;

    public function __construct()
    {
        $this->posts = new ArrayCollection();
        $this->membre = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomGroupe(): ?string
    {
        return $this->nomGroupe;
    }

    public function setNomGroupe(string $nomGroupe): self
    {
        $this->nomGroupe = $nomGroupe;

        return $this;
    }

    public function getSlug(): ?string
    {
        return $this->slug;
    }

    public function setSlug(string $slug): self
    {
        $this->slug = $slug;

        return $this;
    }

    /**
     * @return Collection<int, Post>
     */
    public function getPosts(): Collection
    {
        return $this->posts;
    }

    public function addPost(Post $post): self
    {
        if (!$this->posts->contains($post)) {
            $this->posts[] = $post;
            $post->setGroupPost($this);
        }

        return $this;
    }

    public function removePost(Post $post): self
    {
        if ($this->posts->removeElement($post)) {
            // set the owning side to null (unless already changed)
            if ($post->getGroupPost() === $this) {
                $post->setGroupPost(null);
            }
        }

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }

    public function getCreatedAt(): ?\DateTimeImmutable
    {
        return $this->createdAt;
    }

    public function setCreatedAt(\DateTimeImmutable $createdAt): self
    {
        $this->createdAt = $createdAt;

        return $this;
    }

    public function getIsValid(): ?bool
    {
        return $this->isValid;
    }

    public function setIsValid(bool $isValid): self
    {
        $this->isValid = $isValid;

        return $this;
    }

    public function getIsDeleted(): ?bool
    {
        return $this->isDeleted;
    }

    public function setIsDeleted(bool $isDeleted): self
    {
        $this->isDeleted = $isDeleted;

        return $this;
    }

    public function getBut(): ?string
    {
        return $this->but;
    }

    public function setBut(string $but): self
    {
        $this->but = $but;

        return $this;
    }

    /**
     * @return Collection<int, User>
     */
    public function getMembre(): Collection
    {
        return $this->membre;
    }

    public function addMembre(User $membre): self
    {
        if (!$this->membre->contains($membre)) {
            $this->membre[] = $membre;
        }

        return $this;
    }

    public function removeMembre(User $membre): self
    {
        $this->membre->removeElement($membre);

        return $this;
    }
    public function isValid(): ?bool
    {
        return $this->isValid;
    }
}

File: EspritX_Java-Desktop/EspritX_Java-master/src/edu/esprit/entities/feed/GroupPost.java
package edu.esprit.entities.feed;

import java.time.LocalDateTime;

public class GroupPost {
    private int id;
    private int userId;
    private String nomGroupe;
    private String slug;
    private String image;
    private LocalDateTime createdat;
    private Boolean isValid;
    private Boolean isDeleted;
    private String but;

    public GroupPost() {
    }

    public GroupPost(int id, int userId, String nomGroupe, String slug, String image, String but) {
        this.id = id;
        this.userId = userId;
        this.nomGroupe = nomGroupe;
        this.slug = slug;
        this.image = image;
        this.createdat = createdat;
        this.isValid = isValid;
        this.isDeleted = isDeleted;
        this.but = but;
    }

    public GroupPost(int userId, String nomGroupe, String slug, String image, String but) {
        this.userId = userId;
        this.nomGroupe = nomGroupe;
        this.slug = slug;
        this.image = image;


        this.but = but;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public String getNomGroupe() {
        return nomGroupe;
    }

    public void setNomGroupe(String nomGroupe) {
        this.nomGroupe = nomGroupe;
    }

    public String getSlug() {
        return slug;
    }

    public void setSlug(String slug) {
        this.slug = slug;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public LocalDateTime getCreatedat() {
        return createdat;
    }

    public void setCreatedat(LocalDateTime createdat) {
        this.createdat = createdat;
    }

    public Boolean getValid() {
        return isValid;
    }

    public void setValid(Boolean valid) {
        isValid = valid;
    }

    public Boolean getDeleted() {
        return isDeleted;
    }

    public void setDeleted(Boolean deleted) {
        isDeleted = deleted;
    }

    public String getBut() {
        return but;
    }

    public void setBut(String but) {
        this.but = but;
    }
}
